oper8/controller.py

```diff
@@ -187,6 +203,22 @@ def after_verify(
         """
         return True
 
+    def after_verify_unsuccessful(self, session: Session) -> bool:
```

Comment:
Same point here about a failure flag:
```suggestion
    def after_verify_unsuccessful(self, session: Session, failed: bool) -> bool:
```